server/graphql_schemas/event/schema.py

```diff
@@ -1,65 +1,107 @@
-  class Meta:
-    model = Event
-    filter_fields = {}
-    interfaces = (relay.Node,)
+    class Meta:
+        model = Event
+        filter_fields = {}
+        interfaces = (relay.Node,)
 
 
 class CreateEvent(relay.ClientIDMutation):
     class Input:
-      title = graphene.String(required=True)
-      description = graphene.String(required=True)
-      venue = graphene.String(required=True)
-      date = graphene.String(required=True)
-      time = graphene.String(required=False)
-      featured_image = graphene.String(required=False)
-      social_event_id = graphene.Int(required=False)
+        title = graphene.String(required=True)
+        description = graphene.String(required=True)
+        venue = graphene.String(required=True)
+        date = graphene.String(required=True)
+        time = graphene.String(required=False)
+        featured_image = graphene.String(required=False)
+        social_event_id = graphene.String(required=False)
 
     new_event = graphene.Field(EventNode)
 
     @classmethod
     def mutate_and_get_payload(cls, root, info, **input):
         social_event_id = input.get('social_event_id')
-        social_event = Category.objects.get(id=int(social_event_id))
+        social_event = Category.objects.get(
+            id=from_global_id(social_event_id)[1])
         user = info.context.user
-        googleUser = GoogleUser.objects.get(app_user_id=user.id)
+        googleUser = GoogleUser.objects.first()
         new_event = Event(
-          title=input.get('title'),
-          description=input.get('description'),
-          venue=input.get('venue'),
-          date=input.get('date'),
-          time=input.get('time'),
-          featured_image=input.get('featured_image'),
-          creator=googleUser,
-          social_event=social_event
+            title=input.get('title'),
+            description=input.get('description'),
+            venue=input.get('venue'),
+            date=input.get('date'),
+            time=input.get('time'),
+            featured_image=input.get('featured_image'),
+            creator=googleUser,
+            social_event=social_event
         )
         new_event.save()
 
         return cls(new_event=new_event)
 
 
-class EventQuery(object):
-  event = graphene.Field(EventNode,
-                         id=graphene.Int(),
-                         title=graphene.String())
-  events_list = DjangoFilterConnectionField(EventNode)
+class UpdateEvent(relay.ClientIDMutation):
+
+    class Input:
+        title = graphene.String()
+        description = graphene.String()
+        venue = graphene.String()
+        date = graphene.String()
+        time = graphene.String()
+        featured_image = graphene.String()
+        social_event_id = graphene.Int()
+        event_id = graphene.String(required=True)
+
+    updated_event = graphene.Field(EventNode)
+    action_message = graphene.String()
+
+    def update_instance(instance, args, exceptions=['event_id']):
```

Comment:
Well done @parseendavid but it will be better if we have the `update_instance` function as a utility function in a utils/utility folder in `graphql_schema` app so that it can be reusable.

Reply:
👍